src/Datasource/RepositoryInterface.php

```diff
@@ -65,10 +65,10 @@ public function hasField(string $field): bool;
      * type of search that was selected.
      *
      * @param string $type the type of query to perform
-     * @param array|\ArrayAccess $options An array that will be passed to Query::applyOptions()
+     * @param array $options An array that will be passed to Query::applyOptions()
      * @return \Cake\Datasource\QueryInterface
      */
-    public function find(string $type = 'all', $options = []);
+    public function find(string $type = 'all', array $options = []);
```

Comment:
While this is technically a BC break using `ArrayAccess` never actually worked. We don't have any tests for it and the `$options` eventually get passed to `applyOptions()` which does `ksort()` and iterates on the options, neither of which are possible for an `ArrayAccess` instance.

I updated the argument type for `RepositoryInterface:find()` also to keep things in sync.